### DISCUSSION TITLE
Correct steps of building python wrapper from source

### DISCUSSION THREAD

**Discussion by eric80739** [zxing-cpp/zxing-cpp#765] [Q&A]
I rarely compile .whl or .egg files myself and I hope to get some help.
Operating system: Windows 10
cmake: 3.24.0
c++ compiler: MSVC 19.16
When I cd into wrappers/python/ and run python setup.py install, I encounter an error:
```
Unable to locate zxing source code
```
It seems like the path is not found, so I directly specify the path in CMakeLists.txt:
```
set(CORE_PATH D:/zxing-cpp-master/core)
set(CMAKE_CURRENT_SOURCE_DIR D:/zxing-cpp-master)
```
After this, I encounter errors when running:
```
d:\zxing-cpp-master\core\src\WriteBarcode.h(70): error C2039: 'u8string_view': is not a member of 'std'
```
It seems like the compiler version is incorrect.

Next, I cd into the zxing-cpp-master folder, and execute python wrappers/python/setup.py install, and it compiles and installs successfully:
```
-- The C compiler identification is MSVC 19.16.27045.0
...
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe - skipped
...
Checking Build System
Building Custom Rule D:/zxing-cpp-master/core/CMakeLists.txt
cmake_pch.cxx
...
```
But in the end, there is a warning:
```
warning: install_lib: 'build\lib.win-amd64-3.8' does not exist -- no Python modules to install
```
I can see zxing-cpp installed in pip list, but when I try to import zxingcpp, I get a module not found error.
Please help me, thank you.

**Comment by eric80739** [comment]
it turns out that I have to set ZXING_PYTHON_MODULE in CMakeList.txt to ON
after that running "python wrappers/python/setup.py install" works!

**Comment by axxel** [comment]
> When I cd into wrappers/python/ and run python setup.py install, I encounter an error:

This should have worked. I assume you are trying to build this from a git clone of the repository, not an extracted source tarball?

Inside the `wrappers\python` folder you should see a symbolic link named `core` that points to `..\..\core`. Can you confirm that? The `Unable to locate zxing source code` error means cmake can't find this folder.
